core/src/Kokkos_HostSpace.hpp

```diff
@@ -38,6 +38,7 @@ static_assert(false,
 
 #include "impl/Kokkos_HostSpace_deepcopy.hpp"
 #include <impl/Kokkos_MemorySpace.hpp>
+#include <impl/Kokkos_ZeroMemset_fwd.hpp>
```

Comment:
Why did you not move the default host specialization to a separate header?
Not necessarily saying you need to, wondering what was the reasoning.

Reply:
I was a bit confused by what was considered the default impl. If `HostSpace` is what each backend without a specialization falls back to, I'll add it.

Looks like then only `Serial/` and `SYCL/` are left without this change, might as well add them here?

Other than that, `Kokkos_CopyViews.hpp` has it's own in-house impl of `ZeroMemset`, but the calls to `ZeroMemset` in `Kokkos_CopyViews.hpp` don't actually trigger that impl for me (tested Cuda and Serial). I'm guessing that there is some other configuration where that is used?

Reply:
The default implementation is https://github.com/kokkos/kokkos/blob/56ef02c0cd45b62c39b5a789488467331822499d/core/src/Kokkos_CopyViews.hpp#L1329-L1340 and the calls to `ZeroMemnset` should use the specializations instead if provided. This is what you are observing, right?

Reply:
I think it's a good idea to tackle `Serial` and `SYCL` here as well.

Reply:
Even in `CopyViews`, I observed `HostSpace` being called as default (I incorrectly defined Cuda specialization at first, and `HostSpace` was called).

Reply:
Yes, that's all the same template.